### PR TITLE
Update docs for 2.0.3 and add release checklist

### LINKED ISSUE #99: Develop release checklist
Can be a github wiki page, or a vignette (that will be merged w/pkgdown build workflow).

### PR DESCRIPTION
**Pull request by delosh653**
Updates documentation for new patch release, updates maintainer, and adds release checklist to vignettes.

Fixes #99 (Develop release checklist).

### CODE REVIEW

File: NEWS.md [hunk 1]
```diff
@@ -1,3 +1,14 @@
+# growthcleanr 2.0.3 - 2022-11-01
+
+## Added
+
+- CRAN release checklist now added under Developer Guidelines vignette
```
Comment (dchud) [inline NEWS.md:5]: Could add ticket references here, e.g. `(#99)` at the end. Not critical, though.

File: README.Rmd [hunk 2]
```diff
@@ -102,6 +102,7 @@ The rest of this documentation includes:
   notes and suggestions for running `growthcleanr` with large data sources
 - [Next steps](https://carriedaymont.github.io/growthcleanr/articles/next-steps.html),
   notes on potential enhancements to the pediatric and adult algorithms
+- [Developer Guidelines](https://carriedaymont.github.io/growthcleanr/articles/developer-guidelines.html), advice for contributors to this package, including a CRAN release checklist
```
Comment (dchud) [inline README.Rmd:105]: My editor-brain wants to lower case "Guidelines" in the link text for consistency with the others.

File: vignettes/developer-guidelines.Rmd [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  %\VignetteIndexEntry{Developer Guidelines} %\VignetteEngine{knitr::rmarkdown}
+  %\usepackage[utf8]{inputenc}
+---
+
+## CRAN Release Checklist
+
+When deploying updates to CRAN, here a few updates that you need to check off before doing so:
+
+- Update version number in DESCRIPTION in format major.minor.patch (e.g. 1.2.1)
+- Update documentation with `devtools::document()`
```
Comment (dchud) [inline vignettes/developer-guidelines.Rmd:14]: Does `devtools::document()` handle both building `README.md` from `README.Rmd` and rebuilding the vignettes? I'd been knitting the README manually and calling `pkgdown::build_site()` for the vignettes.

File: README.Rmd (hunk 2 above)
Comment (delosh653) [reply README.Rmd]: done!

File: vignettes/developer-guidelines.Rmd (hunk 3 above)
Comment (delosh653) [reply vignettes/developer-guidelines.Rmd]: No, it doesn't -- it only does package documentation. I've specified that and added provisions to re-knit the readme and call pkgdown to the list (and did it for this commit too!).

**Review by dchud — APPROVED** [review]
👍🏼

**Pull request merged**